generators/utilities/src/main/scala/Simulator.scala

```diff
@@ -83,6 +83,11 @@ object GenerateSimFiles extends App with HasGenerateSimConfig {
   }
   def resources(sim: Simulator): Seq[String] = Seq(
```

Comment:
Why do we edit this instead of just using `addResource`?

Reply:
In emulator.cc, you need certain global variables defined or it won't compile. If there is no SimDRAM and we don't add these files here, we'd get a compile error in verilator because `dramsim` isn't there.

Reply:
It might also work to define `dramsim` as a weak symbol somewhere, so `addResource` can be used for these files.